samtranslator/model/eventsources/push.py

```diff
@@ -1167,36 +1167,51 @@ def _add_auth_to_openapi_integration(self, api, editor):
         # Default auth should already be applied, so apply any other auth here or scope override to default
         api_authorizers = api_auth and api_auth.get("Authorizers")
 
-        if method_authorizer != "NONE" and not api_authorizers:
-            raise InvalidEventException(
-                self.relative_id,
-                "Unable to set Authorizer [{authorizer}] on API method [{method}] for path [{path}] "
-                "because the related API does not define any Authorizers.".format(
-                    authorizer=method_authorizer, method=self.Method, path=self.Path
-                ),
-            )
+        # The IAM authorizer is built-in and not defined as a regular Authorizer.
+        enable_iam_authorizer = api_auth and api_auth.get("EnableIamAuthorizer")
```

Comment:
If `EnabledIamAuthorizer` is `yes` or `"true"` or `FN:if...` this expression would trigger to be true and cause us to enable the `IAM_AUTH`. Somewhere we should have validation on the value of `EnbableIamAuthorizer`. I think @mgrandis is looking/thinking about this area already. I would like to make sure we consider any validation we may need to do on this pr before merging. I don't have the best understanding of what should be done in this area. 

@mgrandis Could you take a look through this and suggest how we might add this so we keep things consistent with what you are working through now?

Reply:
We could maybe go with:
`enable_iam_authorizer = api_auth and api_auth.get("EnableIamAuthorizer", False) is True`

If EnableIamAuthorizer is missing, default to False and check that the value is indeed a Boolean True

Reply:
Works for me